src/components/PreviewOverlay/PreviewOverlay.jsx

```diff
@@ -0,0 +1,112 @@
+import React, { useContext } from "react";
+import { EpisodesContext as Episodes } from "COMPONENTS/Episodes/";
+import { TrailerContext as Trailer } from "COMPONENTS/Episode/";
+import { s } from "./";
+
+export default function PreviewOverlay(props){
+	
+	// PROPS
+	// ---------------------------
+	const {
+		episodeId,
+		index,
+		episodeNo,
+		title,
+		previewId,
+		trailerSrc,
+		prevEpisode, //bound callback
+		nextEpisode, //bound callback
+		closeOverlay //bound callback
+	} = props;
+
+	const {
+		prev: prevTrailer = {},
+		next: nextTrailer = {}
+	} = useContext(Trailer);
+
+
+	// CONTEXT
+	// ----------------------------
+	const {
+		activeIndex,
+		overlayActive,
+		episodeCount
+	} = useContext(Episodes).state;
+
+	// RENDER LOGIC
+	// ----------------------------
+	// handle visibility of active episode and episode preview overlay
+	const isActiveEpisode   = activeIndex === index;
+	const isOverlayActive   = isActiveEpisode && overlayActive;
+	const hidden            = !isOverlayActive;
+	
+	const isFirst           = index === 0;
+	const isLast            = index === episodeCount - 1;
+
+	const { src: prevTrailerSrc } = prevTrailer;
+	const { src: nextTrailerSrc } = nextTrailer;
+
+
+	return (
+		<div
+			id={ previewId }
+			className= { s.container }
+			aria-hidden={ hidden }
+		>
+			<header className={ s.details }>
+				<h2 className={ s.episode }>
+					{ `Episode ${episodeNo}` } 
+				</h2>
+				<h1 className={ s.title }>
+					{ title }
+				</h1>
+				<nav>
+					{/*NEXT AND PREVIOUS BUTTONS*/}
+					<div className={ s.controls }>
+						{ !isFirst && (
+							<a 
+								className={ `${s.prev} ${s.button}` }
+								href={ prevTrailerSrc }
+								aria-label="Previous."
+								onClick={ prevEpisode }
+							>
+								Prev
+							</a>
+						)}
+						{ !isLast && (
+							<a 
+								className={ `${s.next} ${s.button}` }
+								href={ nextTrailerSrc } 
+								onClick={ nextEpisode }
+							>
+								Next
+							</a>
+						)}
+					</div>
+					{/*CLOSE BUTTON*/}
+					<a 
+						className={ s.close }
+						href={ `#${episodeId}` }
+						onClick={ closeOverlay }
+					> 
+						<span className={ s.label }> 
+							Close
+						</span> 
+						<div className={ s.icon }>
+							<div className={ `${s.bar} ${s.bar1}` } />
+							<div className={ `${s.bar} ${s.bar2}` } />
+						</div>
+					</a>
+				</nav>
+			</header>
+			<video className={ s.video }>
+				{ isOverlayActive && (
+					<source 
+						src={trailerSrc}
```

Comment:
now that we have resolution controls via the Client provider, we can start to think about serving assets at the appropriate resolutions - ie. stuff like:

```
const data = {
    preview: {
        poster: "path/to/poster.jpg",
        sources: {
            webm: {
                low: "path/to/preview/01/webm/low.webm,
                medium: "path/to/preview/01/webm/medium.webm,
                high: "path/to/preview/01/webm/high.webm
            },
            mp4: {
                low: "path/to/preview/01/mp4/low.mp4,
                medium: "path/to/preview/01/mp4/medium.mp4,
                high: "path/to/preview/01/mp4/high.mp4
            },
        }
    }
};

let resolution;
if(isSmall)       resolution = "low";
else if(isMedium) resolution = "medium";
else              resolution = "high";

const webm = data.preview.sources.webm[resolution];
const mp4  = data.preview.sources.mp4[resolution];

return (
    <video poster={data.preview.poster}>
        <source src={webm} type="video/webm" />
        <source src={mp4} type="video/mp4" />
    </video>
);
```

This also goes for all the images on the site too - so that can either be part of this pass, or part of the polish pass, as I think there's an argument for both - your call!

Reply:
if you stick to the "low"/"medium"/"high" names as well, you could create a util function to reduce repetition of this - just like:

```
const { webm, mp4 } = getBestSource(data.preview.sources);
```

Reply:
I'll add to the board, doesn't seem appropriate for layout phase :)